src/web/nextui/src/app/components/Navigation.tsx

```diff
@@ -23,26 +26,38 @@ export default function Navigation({
   darkMode: boolean;
   onToggleDarkMode: () => void;
 }) {
-  if (process.env.NEXT_PUBLIC_NO_BROWSING) {
-    return (
-      <Stack direction="row" spacing={2} className="nav">
-        <Logo />
-        <DarkMode darkMode={darkMode} onToggleDarkMode={onToggleDarkMode} />
-      </Stack>
-    );
-  }
-  return (
-    <Stack direction="row" spacing={2} className="nav">
+  const [showInfoModal, setShowInfoModal] = useState<boolean>(false);
+
+  const handleModalToggle = () => setShowInfoModal(!showInfoModal);
```

Comment:
This is fine for now, but generally prefer `setShowInfoModal(prevState => !prevState);`: it guarantees freshness which is important when dealing w/ state that will be updated within the scope of a `useEffect` or at short intervals (which may lead to conflicts).